File: pixlus/code/addons/lib/runtime.py
##
# This is just the general structure
# All code is non-working
##

plugins = {}
def loadPlugin(name="", type=""):
	plugins[name] = type

class PLUGIN:
	type = ""
	block = ""
plugin = PLUGIN

params = {}

def setParams(obj):
	global params
	params = obj

class game:
	class block:
		type = ""
		def setBlock(block=""): return None
		def getBlock(self, x=0, y=0, layer=0): return self

class player:
	class inventory:
		def give(item="", quantity=1):
			"""Add an item to the players invintory, with specified quantity"""
			return None
		def take(item="", quantity=1):
			"""Remove an item to the players invintory, with specified quantity"""
			return None
	def moveToWorld(worldName=""):
		return None

async def delay(seconds=1):
	"""Wait <seconds> before running the next line"""
	return True

# export function sleep(seconds=1) {
# 	let milliseconds = seconds * 1000;
# 	return new Promise(resolve => setTimeout(resolve, milliseconds));
# }

File: pixlus/code/addons/lib/runtime.js
import PlaceBlock from "../../play/behind/world_editor.js";
import { currentWorld, switchWorldTo } from "../../play/main.js";
import { cacheBlock } from "../../play/render.js";

export var params = {}

export function setParams(obj={}){
	params = obj;
}

export var game = {
	block: new class {
		type = params?.block?.type
		setBlock(block="") {
			this.type = params?.block?.type;
			PlaceBlock(block, params.block.pos.x, params.block.pos.y, params.block.pos.layer);
			return null;
		}
		getBlock(x, y, layer) {
			this.type = params?.block?.type;
			if(!x && !y && !layer) {
				return this;
			}
			let worldData = localStorage.getItem(`world.${currentWorld}`);
			worldData = JSON.parse(worldData);
			let blockName = worldData[layer-1][y][x];
			let json = cacheBlock(blockName);
			json.type = blockName;
			json.pos = { x, y, layer };
			return json;
		}
	}
}

export var player = {
	inventory: new class {
		give(item="", quantity=1) {
			return null;
		}
		take(item="", quantity=1) {
			return null;
		}
	},
	moveToWorld: switchWorldTo
}